services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/ApplicationController.kt

```diff
@@ -269,6 +276,15 @@ class ApplicationController(
         }
     }
 
+    fun checkApplicationPermission(hankeTunnus: String?, userId: String) {
+        hankeTunnus.let { tunnus ->
+            hankeService.getHankeId(tunnus).let { id ->
+                if (id == null || !permissionService.hasPermission(id, userId, EDIT_APPLICATIONS))
+                    throw HankeNotFoundException(hankeTunnus)
```

Comment:
Since this throws a HankeNotFoundException, this method is only applicable to creating an application. This should be reflected in the name, so it doesn't get re-used inappropriately.

Reply:
Changed function name.